src/tools/r2rdump/R2RSignature.cs

```diff
@@ -717,7 +718,33 @@ private void ParseType(StringBuilder builder)
                     break;
 
                 case CorElementType.ELEMENT_TYPE_ARRAY:
-                    builder.Append("array");
+                    ParseType(builder);
+                    {
+                        builder.Append('[');
+                        uint rank = ReadUInt();
+                        if (rank != 0)
+                        {
+                            uint sizeCount = ReadUInt(); // number of sizes
+                            for (uint sizeIndex = 0; sizeIndex < sizeCount; sizeIndex++)
```

Comment:
Did you try writing to the output the sizes of the dimensions? I'm wondering how noisy it would be to include them.

Reply:
Good point, fixed in 2nd commit.